app/models/katello/glue/pulp/repo.rb

```diff
@@ -255,11 +255,13 @@ def distribution_ids
       end
 
       def package_group_count
-        self.pulp_repo_facts[:content_unit_counts][:package_group] || 0
+        content_unit_counts = self.pulp_repo_facts[:content_unit_counts]
+        content_unit_counts ? content_unit_counts[:package_group] : 0
       end
 
       def puppet_module_count
-        self.pulp_repo_facts[:content_unit_counts][:puppet_module] || 0
+        content_unit_counts = self.pulp_repo_facts[:content_unit_counts]
+        content_unit_counts ? content_unit_counts[:puppet_module] : 0
       end
```

Comment:
> was it because pulp_repo_facts was nil? or :content_unit_counts was nil?

`:content_unit_counts`.

> and what circumstances caused it?

Any attempted removal of a repo, please see [the bug] (https://bugzilla.redhat.com/show_bug.cgi?id=1166365) for more information.